src/download-pool.js

```diff
@@ -2,38 +2,75 @@ import downloadFile from './download-file';
 
 const POOL_LIMIT = 6;
 
+/**
+ * Groups the elements in an array in subarrays defined by limit.
+ * @example
+ *
+ * splice([1, 2, 3, 4, 5], 3); // [[1, 2, 3], [4, 5]]
+ *
+ * @param {Array} arr
+ * @param {Number} limit
+ * @returns {Array} New array with the subgroups.
+ */
 export const splice = (arr, limit) => {
   const splicedArray = [];
+  const baseArray = arr.slice(0);
```

Comment:
Can we use `Array.from(arr)` instead of `arr.slice(0)`? It's the same thing but the first one is more explicit. I just want to improve legibility. ✌🏻